### PR TITLE
Add analyze_text to make make check happy.

### PR DESCRIPTION
**Pull request by Anteru**
This also fixes a few small bugs:

* Slash uses *.sla as the file ending, not *.sl
* IDL has endelse, not elseelse

### CODE REVIEW

File: pygments/lexers/dotnet.py [hunk 1]
```diff
@@ -684,3 +694,18 @@ class FSharpLexer(RegexLexer):
             (r'"', String),
         ],
     }
+
+    def analyse_text(text):
+        """F# doesn't have that many unique features -- |> for matching
+        seems quite common though, in addition to let/match."""
```
Comment (birkenfeld) [inline pygments/lexers/dotnet.py:700]: let/match/-> are quite common in functional and derived languages, e.g. Haskell or Rust.  I wouldn't use them.

File: pygments/lexers/esoteric.py [hunk 2]
```diff
@@ -48,6 +48,15 @@ class BrainfuckLexer(RegexLexer):
         ]
     }
 
+    def analyse_text(text):
+        """It's safe to assume that any reasonable brainfuck program will have
+        more than two consecutive + signs, which is very uncommon in any
+        other programming language."""
+        if '+++++' in text or '------' in text:
+            return 0.5
+        if '+++' in text:
```
Comment (birkenfeld) [inline pygments/lexers/esoteric.py:57]: `+++` and `---` are common in diff/patch files, which are probably more common than brainfuck :)

The longer `+++++++` and `------` sequences are typical for underlines in readable markup like rst/markdown.

Another signature of brainfuck which has less conflicts is probably `[-]`, i.e. "clear this cell".

File: pygments/lexers/int_fiction.py [hunk 3]
```diff
@@ -516,6 +516,15 @@ def get_tokens_unprocessed(self, text):
         while objectloop_queue:
             yield objectloop_queue.pop(0)
 
+    def analyse_text(text):
+        """We try to find a keyword which seem relatively common, unfortunately
+        there is a decent overlap with Smalltalk keywords otherwise here.."""
+        result = 0
+        if re.match('origsource', text, re.IGNORECASE):
```
Comment (birkenfeld) [inline pygments/lexers/int_fiction.py:523]: `match` vs `search` again?

File: pygments/lexers/configs.py [hunk 4]
```diff
@@ -970,3 +970,9 @@ class SingularityLexer(RegexLexer):
             (r'(.+?(?=^\s*%))|(.*)', using(BashLexer), '#pop'),
         ],
     }
+
+    def analyse_text(text):
+        """This is a quite simple script file, but there are a few keywords
+        which seem unique to this language."""
+        if re.match('osversion|includecmd', text, re.IGNORECASE):
```
Comment (birkenfeld) [inline pygments/lexers/configs.py:977]: `match` is anchored at the beginning of the text, is that intended?

File: pygments/lexers/matlab.py [hunk 5]
```diff
@@ -647,6 +647,15 @@ class OctaveLexer(RegexLexer):
         ],
     }
 
+    def analyse_text(text):
+        """Octave is quite hard to spot from a casual glance, but we'll assume
+        that most examples will have an integer vector or matrix somewhere."""
+        # A \ B is actually quite uncommon outside of Matlab/Octave
+        if re.match(r'\w+\s*\\\s*\w+',text):
+            return 0.05
+        if re.match(r'\[\s*(?:(?:\d+\s*)+;\s*)+\s*\]', text):
```
Comment (birkenfeld) [inline pygments/lexers/matlab.py:656]: This is a quite complicated regex that might backtrack (not catastrophically, but still a bit much to do for analyse_text...)

File: pygments/lexers/modula2.py [hunk 6]
```diff
@@ -1559,3 +1559,13 @@ def get_tokens_unprocessed(self, text):
 
             # return result
             yield index, token, value
+
+    def analyse_text(text):
+        """Not much we can go by. (* for comments is our best guess."""
```
Comment (birkenfeld) [inline pygments/lexers/modula2.py:1564]: that's the same for all pascal-like languages, unfortunately.

File: pygments/lexers/scripting.py [hunk 7]
```diff
@@ -944,6 +944,15 @@ class HybrisLexer(RegexLexer):
         ],
     }
 
+    def analyse_text(text):
+        """public method and private method don't seem to be quite common
+        elsewhere."""
+        result = 0
+        if re.match(r'(?:public|private)\s+method', text):
```
Comment (birkenfeld) [inline pygments/lexers/scripting.py:951]: `match`

**Review by birkenfeld — CHANGES_REQUESTED** [review]
In general I'd prefer for searched-for keywords to be bounded by a `\b` word delimiter.

File: pygments/lexers/configs.py (hunk 4 above)
Comment (Anteru) [reply pygments/lexers/configs.py:977]: Nope, and I'll fix all remaining instances as well.

File: pygments/lexers/matlab.py (hunk 5 above)
Comment (Anteru) [reply pygments/lexers/matlab.py:656]: I just tried those out and both aren't really good indicators either. I'm inclined to `return 0` until someone has a better idea.

File: pygments/lexers/esoteric.py (hunk 2 above)
Comment (Anteru) [reply pygments/lexers/esoteric.py]: I've changed it to check if 25% of the source consist of +- or <>, that seems fairly unlikely in any other source (even HTML is not that <> heavy.)

Comment (birkenfeld) [reply pygments/lexers/esoteric.py]: While this is probably a good test, I'm worrying about the performance implications of looping over the string on the Python level... the other tests at least use the C-based regex/stringlib routines.

Comment (Anteru) [reply pygments/lexers/esoteric.py]: Given it's statistics, maybe trying it on the first 256 characters or so is enough? The character distribution for brainfuck should be similar across any subset of the program.

File: pygments/lexers/scripting.py (hunk 7 above)
Comment (Anteru) [reply pygments/lexers/scripting.py]: Done

File: pygments/lexers/modula2.py (hunk 6 above)
Comment (Anteru) [reply pygments/lexers/modula2.py:1564]: Ok. New plan: Figure out if it's Pascal-Like first, then check if `PROCEDURE` is present (not valid Pascal), and back out if `FUNCTION` is found (not valid Modula2)

File: pygments/lexers/ezhil.py [hunk 8]
```diff
@@ -69,7 +69,7 @@ def analyse_text(text):
         decent amount of Tamil-characters, it's this language. This assumption
         is obviously horribly off if someone uses string literals in tamil
         in another language."""
-        if len(re.findall('[\u0b80-\u0bff]')) > 10:
+        if len(re.findall(r'[\u0b80-\u0bff]', text)) > 10:
```
Comment (birkenfeld) [inline pygments/lexers/ezhil.py:72]: Oops, I missed that.

File: pygments/lexers/modula2.py [hunk 9]
```diff
@@ -1563,9 +1563,20 @@ def get_tokens_unprocessed(self, text):
...
+
+        is_pascal_like = 0
         if '(*' in text and '*)' in text:
-            result += 0.01
+            is_pascal_like += 0.5
         if ':=' in text:
-            result += 0.01
+            is_pascal_like += 0.5
+
+        if is_pascal_like == 1:
```
Comment (birkenfeld) [inline pygments/lexers/modula2.py:1573]: Looks good, but can you make this a boolean flag (`pascal = x in text and y in text and z in text`)?  The float-adding-and-comparing seems to be unneeded...

**Review by birkenfeld — APPROVED** [review]

File: pygments/lexers/ezhil.py (hunk 8 above)
Comment (Anteru) [reply pygments/lexers/ezhil.py:72]: Me too, but that's what ~friends~ tests are for :)

File: pygments/lexers/modula2.py (hunk 9 above)
Comment (Anteru) [reply pygments/lexers/modula2.py:1573]: Sure!

**Pull request merged**